### PR TITLE
feat(audio-only): be able to lock a browser into capturing audio only

### PR DESCRIPTION
**Pull request by virtuacoplenny**

### CODE REVIEW

File: JitsiMediaDevices.js [hunk 1]
```diff
@@ -138,6 +138,18 @@ const JitsiMediaDevices = {
...
         eventEmitter.emit(event, ...args);
+    },
+
+    /**
+     * Returns whether or not the current browser is limited to supporting only
+     * audio send and receive.
+     *
+     * @returns {boolean}
+     */
+    supportsOnlyAudio() {
```
Comment (bbaldino) [inline JitsiMediaDevices.js]: how about inverting this and doing `supportsVideo`

Comment (virtuacoplenny) [reply JitsiMediaDevices.js]: supportVideoOnly or separate checks for supporting audio and supporting video?

Comment (bbaldino) [reply JitsiMediaDevices.js]: yeah, could make it totally dynamic and have both

File: JitsiMediaDevices.js [hunk 2]
```diff
@@ -138,6 +138,18 @@ const JitsiMediaDevices = {
...
+    /**
+     * Returns whether or not the current browser can support capturing video,
+     * be it camera or desktop, and displaying received video.
+     *
+     * @returns {boolean}
+     */
+    supportsVideo() {
+        // Defer to RTCBrowserType to allow exposure of the api to the consumer
+        // but prevent other files from having to import JitsiMediaDevices.
+        return RTCBrowserType.supportsVideo();
```
Comment (bbaldino) [inline JitsiMediaDevices.js]: can we just expose RTCBrowserType?

Comment (damencho) [reply JitsiMediaDevices.js]: I remember we had a discussion at some point about exporting RTCBrowserType so we can use it in jitsi-meet. We decided not to do that ... but I don't remember details.

File: modules/RTC/RTCUtils.js [hunk 3]
```diff
@@ -1532,7 +1532,15 @@ class RTCUtils extends Listenable {
         const maybeRequestCaptureDevices = function() {
             const umDevices = options.devices || [ 'audio', 'video' ];
             const requestedCaptureDevices = umDevices.filter(
-                device => device !== 'desktop');
+                device => {
```
Comment (bbaldino) [inline modules/RTC/RTCUtils.js]: how about
```
umDevices
  .filter(d => {
    if (d === 'audio') return true;
    if (d === 'video' && RTCVideoBrowserType.supportsVideo) return true;
    return false;
  });
```

**Comment by virtuacoplenny** [comment]
Rebased onto latest master to get package-lock, squashed previous commits for addressing feedback, and force pushed.

**Pull request merged**